api/src/pcapi/core/users/api.py

```diff
@@ -663,6 +665,17 @@ def update_user_info(
             snapshot.set("idPieceNumber", old=user.idPieceNumber, new=id_piece_number)
         user.idPieceNumber = id_piece_number
 
+    if user.pro_new_nav_state and new_nav_date is not UNCHANGED:
+        user.pro_new_nav_state.newNavDate = new_nav_date if new_nav_date is not None else None
+    if user.pro_new_nav_state and eligibility_date is not UNCHANGED:
+        user.pro_new_nav_state.eligibilityDate = eligibility_date if eligibility_date is not None else None
+    if user.pro_new_nav_state is None and (new_nav_date is not UNCHANGED or eligibility_date is not UNCHANGED):
+        user.pro_new_nav_state = models.UserProNewNavState(
```

Comment:
```suggestion
       pro_new_nav_state = models.UserProNewNavState(...)
       db.session.add(pro_new_nav_state)
       user.pro_new_nav_state = pro_new_nav_state
```

Dans une procaine version, sqlalchemy n'ajoutera plus de façon implicite les relations à la session courante